modules/javafx.graphics/src/test/java/test/javafx/scene/Node_lookup_Test.java

```diff
@@ -96,4 +109,131 @@ public class Node_lookup_Test {
+
+    @Test
+    public void lookupPseudoTest() {
+        Set<Node> nodes = root.lookupAll(".h:testPseudo2");
+        assertEquals(1, nodes.size());
+        assertTrue(nodes.contains(hg));
+
+        Node found = root.lookup(".h:testPseudo2");
+        assertSame(hg, found);
+
+        found = root.lookup(":testPseudo2");
+        assertSame(hg, found);
+
+        found = root.lookup(".h:testPseudo1:testPseudo2");
+        assertSame(hg, found);
+
+        nodes = root.lookupAll(".g:testPseudo1");
+        assertEquals(2, nodes.size());
+        assertTrue(nodes.contains(g));
+        assertTrue(nodes.contains(hg));
+
+        nodes = root.lookupAll(":testPseudo1");
+        assertEquals(2, nodes.size());
+        assertTrue(nodes.contains(g));
+        assertTrue(nodes.contains(hg));
+
+        nodes = root.lookupAll(".f > .h:testPseudo1");
+        assertEquals(1, nodes.size());
+        assertTrue(nodes.contains(hg));
+
+        nodes = root.lookupAll(".f:testPseudo > .h:testPseudo1");
+        assertEquals(1, nodes.size());
+        assertTrue(nodes.contains(hg));
+
+        nodes = root.lookupAll(".f:randomPseudo");
+        assertEquals(0, nodes.size());
+    }
+
+    /**
+     * Verifies that the lookup ignores pseudo classes when selector contains no explicit pseudo class, but all the nodes have pseudo classes set to them.
+     */
+    @Test
+    public void lookupPseudoTest2() {
+        // Except root node all the other nodes (f, g, hg) have pseudo classes set to them
+        Set<Node> nodes = root.lookupAll(".g");
+        assertEquals(2, nodes.size());
+        assertTrue(nodes.contains(g));
+        assertTrue(nodes.contains(hg));
+
+        nodes = root.lookupAll("#root .g");
+        assertEquals(2, nodes.size());
+        assertTrue(nodes.contains(g));
+        assertTrue(nodes.contains(hg));
+
+        nodes = root.lookupAll(".f .g");
+        assertEquals(2, nodes.size());
+        assertTrue(nodes.contains(g));
+        assertTrue(nodes.contains(hg));
+
+        nodes = root.lookupAll(".f .h");
+        assertEquals(1, nodes.size());
+        assertTrue(nodes.contains(hg));
+
+        nodes = root.lookupAll(".f > .h");
+        assertEquals(1, nodes.size());
+        assertTrue(nodes.contains(hg));
+
+        nodes = root.lookupAll(".h");
+        assertEquals(1, nodes.size());
+        assertTrue(nodes.contains(hg));
+
+        nodes = root.lookupAll("#root > .f");
+        assertEquals(1, nodes.size());
+        assertTrue(nodes.contains(f));
+
+        nodes = root.lookupAll("#root .f");
+        assertEquals(1, nodes.size());
+        assertTrue(nodes.contains(f));
+
+        nodes = root.lookupAll(".random");
+        assertEquals(0, nodes.size());
+
+        nodes = root.lookupAll(".random .h");
+        assertEquals(0, nodes.size());
+    }
+
+    /**
+     * Verifies that the lookup ignores pseudo classes when selector contains no explicit pseudo class.
+     */
+    @Test
+    public void lookupPseudoTest3() {
```

Comment:
Consider naming the test methods according to what they test.

Reply:
Renamed the methods by referring to method names used in other classes. I noticed that descriptive long method names is accepted, so named similarly. The only thing I want to check is: I renamed existing methods also. Is it accepted? or I am not allowed to rename others method names?

Reply:
I see no problem renaming: there are no dependencies.